### PR TITLE
minecraft uses standard coordinate system

### PR DESCRIPTION
**Pull request by aszlam**
# Description
This moves craftassist onto the same coordinate system as on the robots inside the droidlet agent (minecraft still retains its unusual coordinate system :| ).  Droidlet should all live in same coordinates from now on...

In addition hand updated several gt parses

## Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [x] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

### CODE REVIEW

File: agents/craftassist/craftassist_agent.py [hunk 1]
```diff
@@ -403,31 +410,31 @@ def npy_to_blocks_list(npy, origin):
         sio.emit("updateVoxelWorldState", payload)
 
     def step_pos_x(self):
-        self.cagent.step_pos_x()
+        self.mover.step_pos_x()
         self.update_agent_pos_dashboard()
 
     def step_neg_x(self):
-        self.cagent.step_neg_x()
+        self.mover.step_pos_x()
```
Comment (soumith) [inline agents/craftassist/craftassist_agent.py:417]: step_neg_x returns step_pos_x?

**Review by soumith — APPROVED** [review]

**Comment by aszlam** [comment]
@kavyasrinet @snyxan @ethan-carlson if it is possible, I would like a serious review on this.

**Comment by aszlam** [comment]
@soumith I made one more change after your previous review that affects the robot, see https://github.com/facebookresearch/fairo/blob/update_rotations/droidlet/shared_data_struct/rotation.py#L97.  Previously yaw_pitch and look_vec were not agreeing on the sign of pitch; now they should agree and match the figure in the file (+pitch faces down, -pitch faces up).  This affects robot via https://github.com/facebookresearch/fairo/blob/update_rotations/droidlet/lowlevel/robot_mover_utils.py#L47 and then for example https://github.com/facebookresearch/fairo/blob/update_rotations/droidlet/lowlevel/hello_robot/hello_robot_mover.py#L179   
but I think mostly does not affect robot proper.

**Comment by aszlam** [comment]
question for everyone:  droidlet coordinate system here has pitch - facing up and pitch + facing down.  That is to me counterintuitive but also how both minecraft and the hello robot are set up (going from memory rn on hello, at home, will check on monday).  should we stick with this convention?

**Comment by aszlam** [comment]
> question for everyone: droidlet coordinate system here has pitch - facing up and pitch + facing down. That is to me counterintuitive but also how both minecraft and the hello robot are set up (going from memory rn on hello, at home, will check on monday). should we stick with this convention?

ok, this turns out to be not true- Hello has positive pitch turning up.  I am going to change MC to follow this, and uniformize around +pitch looks up, -pitch looks down inside droidlet agent

**Review by ethan-carlson — APPROVED** [review]
This makes sense to me.  I tested relative directions and the commands were all parsed and executed as expected.  We haven't connected vision error marking with annotation yet and I already have a task to verify that position and look line up throughout the pipeline, so this doesn't break anything on main.

**Comment by kavyasrinet** [comment]
> > question for everyone: droidlet coordinate system here has pitch - facing up and pitch + facing down. That is to me counterintuitive but also how both minecraft and the hello robot are set up (going from memory rn on hello, at home, will check on monday). should we stick with this convention?
> 
> ok, this turns out to be not true- Hello has positive pitch turning up. I am going to change MC to follow this, and uniformize around +pitch looks up, -pitch looks down inside droidlet agent

I agree with this - this is more intuitive and easier to maintain / understand as well.

File: agents/craftassist/tests/base_craftassist_test_case.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 from droidlet.memory.craftassist.mc_memory_nodes import VoxelObjectNode
 from droidlet.lowlevel.minecraft.mc_util import XYZ, Block, IDM
-from droidlet.perception.craftassist.rotation import yaw_pitch
+from droidlet.shared_data_struct.rotation import yaw_pitch
```
Comment (kavyasrinet) [inline agents/craftassist/tests/base_craftassist_test_case.py:11]: this is great

File: droidlet/interpreter/facing_utils.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    a left turn, or counterclockwise viewed from above has a negative degree value
+    a right turn, or clockwise viewed from above has a positive degree value
+
+    TODO specify this with a const to be imported
+    """
+    left = "left" in d["relative_yaw"] or "leave" in d["relative_yaw"]  # lemmatizer :)
+    if left or "right" in d["relative_yaw"]:
+        # don't allow negative values when a direction is specified, FIXME?
+        degrees = abs(number_from_span(d["relative_yaw"])) or 90
+        if left:
```
Comment (kavyasrinet) [inline droidlet/interpreter/facing_utils.py:34]: I think we should document this in README somewhere- left is +ve degree, right is negative, positive and negative yaw and pitch for up and down as well

File: droidlet/lowlevel/minecraft/craftassist_mover.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    if getattr(struct, "x", None):
+        return flip_x(struct)
+    elif getattr(struct, "yaw", None):
+        return flip_look(struct)
+    elif getattr(struct, "mobType", None):
+        # we keep the cagent structas an attribute in case we want to interface with cagent again
+        return Mob(
+            struct.entityId, struct.mobType, flip_x(struct.pos), flip_look(struct.look), struct
+        )
+    elif getattr(struct, "mainHand", None):
```
Comment (kavyasrinet) [inline droidlet/lowlevel/minecraft/craftassist_mover.py:42]: is mainHand an attribute of Player struct ? Can we add a comment

File: droidlet/lowlevel/minecraft/craftassist_mover.py [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+    def step_neg_z(self):
+        self.cagent.step_neg_z()
+
+    def step_forward(self):
+        self.cagent.step_forward()
+
+
+############################################################################
+# in minecraft, we have
```
Comment (kavyasrinet) [inline droidlet/lowlevel/minecraft/craftassist_mover.py:188]: Ah just saw that the documentation is in this file, thanks for adding a mover. Might still be worth linking this file from main README for ca agent

File: droidlet/perception/robot/tests/utils.py [hunk 6]
```diff
@@ -23,7 +24,8 @@ def init_physical_interfaces(self):
 
     def init_perception(self):
         self.perception_modules = {}
-        self.perception_modules["language_understanding"] = NSPQuerier(self.opts, self)
+        self.perception_modules["language_understanding"] = None
```
Comment (kavyasrinet) [inline droidlet/perception/robot/tests/utils.py:27]: was commenting this out intentional or just meant for local testing ?

**Comment by kavyasrinet** [comment]
@snyxan these changes will also affect voxel world work and will break things upstream on Turk - tagging you here to closely review the coordinate change as well

File: droidlet/perception/robot/tests/utils.py (hunk 6 above)
Comment (aszlam) [reply droidlet/perception/robot/tests/utils.py:27]: it was just for local testing, I guess I forgot to fix this one.   but!  I think this whole object is dead code, we are not ever using this FakeAgent

**Pull request merged**